### PR TITLE
Fix #926 cache warmer and method signature

### LINKED ISSUE #926: Cache warmer is broken in v0.15.0
**Actual behavior**
Cache warmer never downloads images that are not found in the cache

**Expected behavior**
cache warmer should download images not found in the cache

**To Reproduce**
1. `docker run -v $(pwd):/workspace gcr.io/kaniko-project/warmer:latest --cache-dir=/workspace/cache --image=ubuntu:latest`

**Additional Information**
 This commit broke it https://github.com/GoogleContainerTools/kaniko/commit/a6e458caf1aca5deda542269714cde53becf1544
 
 **Triage Notes for the Maintainers**
 

 
 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

### PR DESCRIPTION
**Pull request by cvgw**
Fixes #926 .

**Description**

A previous change had broken the cache warmer by updating the return signature of a method which the cache warmer called.

This fixes that issue, refactors the warming code for easier testing, and adds unit tests.

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [x] The code flow looks good. 
- [x] Unit tests and or integration tests added.

**Release Notes**

Describe any changes here so maintainer can include it in the release notes, or delete this block.

```
- Fix cache warmer error
```

### CODE REVIEW

File: hack/boilerplate/boilerplate.py [hunk 1]
```diff
@@ -22,142 +22,156 @@
 import re
 import sys
 
-
-SKIPPED_DIRS = ["Godeps", "third_party", ".git", "vendor", "examples", "testdata"]
+SKIPPED_DIRS = [
```
Comment (tejal29) [inline hack/boilerplate/boilerplate.py]: Looks like this is irrelevant to this Change. can we move this out in a different PR

Comment (cvgw) [reply hack/boilerplate/boilerplate.py]: The file was autoformated by `google/codefmt-vim` (which I think is uses `Autopep8` for python) when I updated the list of valid years. I can revert this file, disable the formatter, and redo the change to years if you prefer

Comment (cvgw) [reply hack/boilerplate/boilerplate.py]: @tejal29 I reverted this and submitted the change as #949 instead
